add Twitter Bootstrap with LESS

cyborgsimon opened the pull request:
nemo64:bootstrap will be installed if this option is selected
added tests for the cssPreprocessor prompt
fixed gitignore for the Temporary test directory
added longer descriptions for the cssPreprocessor options

cyborgsimon committed:
add Twitter Bootstrap with LESS as option

nemo64:bootstrap will be installed if this option is selected
added tests for the cssPreprocessor prompt
fixed gitignore for the Temporary test directory
added longer descriptions for the cssPreprocessor options

cyborgsimon committed:
improved packages template and added test for none

cmalven commented:
Even the default version of Bootstrap uses Less, so I don't think "LESS version of Twitter Bootstrap" is a very accurate description of how the nemo64 version is different.

I hardly ever use bootstrap so I don't feel strongly about keeping the default Bootstrap as an option if you think the nemo64 version is dramatically better.

cmalven commented:
Oh, BTW @cyborgsimon I also renamed the project (`meteor-coffee` is much more of a pain to type than `met` when running generators) so you'll probably need to take that into account in your code.

cyborgsimon commented:
hmm i liked the version of nemo64 because you can select which parts you need. so if it is ok with you i will use it for the bootstrap option and remove the other option.

renaming the project is a good idea. let me do it asap so you do not need to merge 2 times.

cyborgsimon committed:
only use the nemo64:bootstrap package for twitter

removed the second twitter option

cyborgsimon commented:
any other suggestions?

cmalven commented:
I had one more idea @cyborgsimon: since it seems like nemo64:bootstrap requires you to create a `custom.bootstrap.json` file before it works, we should probably either instruct people to create this file as a Yeoman prompt or create it for them if they select the Bootstrap option.

cyborgsimon committed:
Merge remote-tracking branch 'cmalven/master'

cyborgsimon committed:
Merge branch 'master' into bootstrap-less

cyborgsimon committed:
let user choose between the most common bootstrap packages

add twbs:bootstrap as option
improved nemo64:bootstrap option
- add custom.bootstrap.json with all options enabled
- add styles.less template to import generated bootstrap LESS files
- add tests
- fix issues with renaming meteor-coffee generator to met generator

cyborgsimon commented:
added also twbs:bootstrap as option since it seams to be the most used one
added custom.bootstrap.json as requested
fixed some renaming issues
added tests

cmalven commented on app/templates/meteor/_packages:
```diff
@@ -4,10 +4,9 @@
-stylus<% } %><% if (cssPreprocessor == 'less') { %>
-less<% } %><% if (cssPreprocessor == 'bootstrap') { %>
-bootstrap<% } %>
+coffeescript<% if (cssPreprocessor == 'nemo64:bootstrap') { %>
```
Maybe I'm wrong, but won't the `less` package automatically be included in the project if we use one of the bootstrap packages (since they include Less as a dependency)?

cyborgsimon commented:
Did not test. On the GitHub page (https://github.com/Nemo64/meteor-bootstrap) it states:
1. execute meteor add nemo64:bootstrap less

cmalven commented:
Checked out https://github.com/Nemo64/meteor-bootstrap/blob/master/package.js

Surprisingly, they don't specify the `less` package as a dependency. That's very strange to me, but maybe they have a good reason for it. Nevermind!

cyborgsimon commented:
i guess they will compile it to CSS if LESS is not available, but not 100% sure (did not look at the code, but a little voice in my head whispered that it may have read something about it).
